Source/WebCore/dom/SimpleRange.h

```diff
@@ -29,6 +29,47 @@
 
 namespace WebCore {
 
+struct SimpleRangeWeak {
+    BoundaryPointWeak start;
+    BoundaryPointWeak end;
+
+    std::optional<std::reference_wrapper<Node>> startContainer() const
+    {
+       if (!start.container)
+            return std::nullopt;
+        return *start.container;
+    }
+
+    std::optional<Ref<Node>> protectedStartContainer() const
+    {
+        if (!start.container) {
+            return std::nullopt;
+        }
+        return Ref { *start.container };
+    }
+
+    unsigned startOffset() const { return start.offset; }
+
+    std::optional<std::reference_wrapper<Node>> endContainer() const
+    { 
+       return startContainer();
+    }
+
+    std::optional<Ref<Node>> protectedEndContainer() const
+    {
+        return protectedStartContainer();
+    }
+    unsigned endOffset() const { return end.offset; }
+
+    bool collapsed() const { return start == end; }
+
+    friend bool operator==(const SimpleRangeWeak&, const SimpleRangeWeak&) = default;
+
+    SimpleRangeWeak(const BoundaryPointWeak&, const BoundaryPointWeak&);
+    SimpleRangeWeak(BoundaryPointWeak&&, BoundaryPointWeak&&);
+    SimpleRangeWeak(const BoundaryPoint&&, const BoundaryPoint&&);
+};
```

Comment:
For a struct like this, do we typically make a new .h and .cpp file for it? Or, do we try not to make new files to avoid increasing compilation times?